specs/plasma.md

```diff
@@ -0,0 +1,188 @@
+
+In addition, we filter events from the DA Challenge contract included in the block
+and sync a local state of challenged input commitments. As the derivation pipeline steps through
+`challenge_window + resolve_window` amount of L1 blocks, any challenge marked as active
+becomes expired causing a reset of the derivation pipeline.
+
+```js
+// The status enum of a DA Challenge event
+enum ChallengeStatus {
+    Uninitialized,
+    Active,
+    Resolved,
+    Expired
+}
+
+// DA Challenge event filtered
+event ChallengeStatusChanged(
+  bytes32 indexed challengedHash, uint256 indexed challengedBlockNumber, ChallengeStatus status
+);
+
+```
+
+Derivation can either be driven by new L1 blocks or restarted (reset) from the L1 origin of the last
+L2 safe head known by the execution engine.
+
+When stepping through new L1 origin blocks, input data is loaded from the DA storage service.
+If the service responds with a 404 (not found) error, derivation stalls and the DA Manager
+starts a detached traversal of new L1 blocks until:
+
+- An `Active` challenge event is found, detached L1 traversal continues:
+  - A `Resolved` challenge event is found: derivation continues with the input data submitted onchain.
+  - A high enough L1 block is reached such as `latest_l1 - pipeline_l1_origin = resolve_window_size`:
+    input data is skipped from the derivation output.
+- Data is never challenged, derivation returns critical error.
+
+When derivation is reset, the pipeline steps through previously seen L1 origins so that
+`block_start = latest_l1_block - (challenge_window_size + resolve_window_size + sequencer_window_size)`.
+In that case, the DA manager has already synced the state of challenges during the range of L1 blocks traversed
+so the pipeline can skip commitments with expired challenges and reorg the L2 chain
+or load input data from the resolving transaction calldata.
+
+## Safety and Finality
+
+Similarly to rollup mode, the engine queue labels any new blocks derived from input data with a commitment
+on the L1 chain as “safe”. Although labeled as “safe”, the chain might still reorg in case of a faulty DA provider
+and users must use the “finalized” label for a guarantee that their state cannot revert.
+
+With Plasma mode on, the engine queue does not receive finality signals from the L1 RPC
+but from the DA manager that keeps track of challenges. The engine queue will maintain a longer buffer
+of L2 blocks waiting for the DA windows to expire in order to be finalized.
+
+## Security Considerations
+
+The Data Availability Challenge contract mitigates DoS vulnerability with a payable bond requirement making
+challenging the availability of a commitment at least as expensive as submitting the data onchain to resolve
+the challenge.
+In addition, the reward is not net positive for the fisherman who forced the release of data
+by challenging thus preventing money pump vulnerability while still making challenging affordable to altruistic
```

Comment:
No definition of fishermen here makes it very contextual, perhaps link to EF research?